odoo/models.py

```diff
@@ -2544,6 +2544,7 @@ def _inherits_check(self):
                 if field.ondelete.lower() not in ('cascade', 'restrict'):
                     field.ondelete = 'cascade'
                 self._inherits[field.comodel_name] = field.name
+                self.env.registry[field.comodel_name]._inherits_children.add(self._name)
```

Comment:
```suggestion
                self.pool[field.comodel_name]._inherits_children.add(self._name)
```
is 100% equivalent and more concise.